build_tools/aws-sdk-code-generator/lib/aws-sdk-code-generator/views/endpoints_module.rb

```diff
@@ -134,6 +130,8 @@ def built_in_client_context_param_value(param_data)
           else
             'context.config.use_dualstack_endpoint'
           end
+        when 'AWS::Auth::AccountId'
+          "context.config.account_id_endpoint_mode == 'disabled' ? nil : context.config.credentials.credentials.account_id"
```

Comment:
Since this logic may end up pushed into endpoint rules anyway, this probably doesn't matter - but I assume here is where we would need to warn/raise for preferred/required when account id is not set. (or rather, generate the code to do so)

Reply:
If the logic is pushed to rules, yeah, all we do is pass in the endpoint mode. I don't think we care about warning (can discuss), and raising would be done by rules, right?